jax/_src/scipy/optimize/bfgs.py

```diff
@@ -133,7 +136,15 @@ def body_fun(state):
         failed=line_search_results.failed,
         line_search_status=line_search_results.status,
     )
-    s_k = line_search_results.a_k * p_k
+    alpha_k = line_search_results.a_k
+    # If the step is too small in 32 bit mode then the optimizer can get stuck
+    # with a direction of zero - ensure that doesn't happen by setting a
+    # minimum step size.
+    if not jax.config.x64_enabled:
```

Comment:
I'm not 100% sure if the input type matters - because the jax arrays which have precision issues are all created inside the function, rather than the input arrays.

As another datapoint the test I wrote to test passing in a float32 input didn't compile, because the output type from the loop (float64 array created inside the function) was different to the input type (float32 array created outside).

Based on that I'd be inclined to leave it as it is, but obviously defer to you if you would prefer testing the input type?